giskard/core/dataset_validation.py

```diff
@@ -8,18 +8,20 @@
 from giskard.datasets.base import Dataset
 
 
-def validate_target(ds: Dataset):
+def validate_optional_target(ds: Dataset):
     if not ds.target:
         warning(
             "You did not provide the optional argument 'target'. "
             "'target' is the column name in df corresponding to the actual target variable (ground truth)."
         )
-    else:
-        if ds.target not in ds.columns:
-            raise ValueError(
-                "Invalid target parameter:"
-                f" '{ds.target}' column is not present in the dataset with columns: {list(ds.columns)}"
-            )
+
+
+def validate_target_exists(ds: Dataset):
+    if ds.target and ds.target not in ds.columns:
```

Comment:
We must check explicitly for the value of `ds.target`. For example, 0 should be a valid target name (as far as pandas is concerned) but it would evaluate to False here.

Reply:
Explicit check for `ds.target is None`

Reply:
agreed